From bb71b21eb9407479ad729b3c858b0fc350fae335 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Sun, 21 Feb 2021 19:39:59 +0100
Subject: [PATCH] CMakeLists.txt: add BUILD_EXAMPLE

Allow the user to disable example

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[yann.morin.1998@free.fr: backport from upstream]
Signed-off-by: Yann E. MORIN <yann.morin.1998@free.fr>
---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 110d933..c52ab79 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,11 @@ cmake_minimum_required(VERSION 2.8)
 
 project(libuwsc C)
 
+option(BUILD_EXAMPLE "Build example" ON)
+
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/Modules/")
 
 add_subdirectory(src)
-add_subdirectory(example)
+if(BUILD_EXAMPLE)
+    add_subdirectory(example)
+endif()
-- 
2.25.1

